### PR TITLE
br: update new_collation related logs

### LINKED ISSUE #46136: config show new_collations_enabled_on_first_bootstrap result not correct after upgrade from v5.4.1 to v6.5.3
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

TiUP deploy a new v5.4.1 version cluster. yaml file no need set new_collations_enabled_on_first_bootstrap.

mysql> show config where name like '%new_collations_enabled_on_first_bootstrap%';
+------+---------------+-------------------------------------------+-------+
| Type | Instance      | Name                                      | Value |
+------+---------------+-------------------------------------------+-------+
| tidb | 10.x.x.3:4923 | new_collations_enabled_on_first_bootstrap | false |
| tidb | 10.x.x.3:4913 | new_collations_enabled_on_first_bootstrap | false |
+------+---------------+-------------------------------------------+-------+

mysql> select * from mysql.tidb;
+-----------------------+--------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------+
| VARIABLE_NAME         | VARIABLE_VALUE                                                                                   | COMMENT                                                                                |
+-----------------------+--------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------+
| bootstrapped          | True                                                                                             | Bootstrap flag. Do not delete.                                                         |
| tidb_server_version   | 83                                                                                               | Bootstrap version. Do not delete.                                                      |
| system_tz             | Asia/Shanghai                                                                                    | TiDB Global System Timezone.                                                           |
| new_collation_enabled | False                                                                                            | If the new collations are enabled. Do not edit it.                                     |

upgrade v5.4.1 version to v6.5.3 version.

show config where name like '%new_collations_enabled_on_first_bootstrap%';
+------+---------------+-------------------------------------------+-------+
| Type | Instance      | Name                                      | Value |
+------+---------------+-------------------------------------------+-------+
| tidb | 10.x.x.3:4913 | new_collations_enabled_on_first_bootstrap | true  |
| tidb | 10.x.x.3:4923 | new_collations_enabled_on_first_bootstrap | true  |
+------+---------------+-------------------------------------------+-------+

 select * from mysql.tidb;
+-----------------------+--------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------+
| VARIABLE_NAME         | VARIABLE_VALUE                                                                                   | COMMENT                                                                                |
+-----------------------+--------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------+
| bootstrapped          | True                                                                                             | Bootstrap flag. Do not delete.                                                         |
| tidb_server_version   | 91                                                                                               | Bootstrap version. Do not delete.                                                      |
| system_tz             | Asia/Shanghai                                                                                    | TiDB Global System Timezone.                                                           |
| new_collation_enabled | False                                                                                            | If the new collations are enabled. Do not edit it.                                     |

### 2. What did you expect to see? (Required)

show config where name like '%new_collations_enabled_on_first_bootstrap%';

The result should be false. 

The show config result should show the true value of the new_collations_enabled_on_first_bootstrap parameter.(not default value)

### 3. What did you see instead (Required)

show config where name like '%new_collations_enabled_on_first_bootstrap%';

The result is true

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #46136

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/restore/client.go [hunk 1]
```diff
@@ -3689,12 +3689,12 @@ func CheckNewCollationEnable(
 	if backupNewCollationEnable == "" {
 		if CheckRequirements {
 			return errors.Annotatef(berrors.ErrUnknown,
-				"the config 'new_collations_enabled_on_first_bootstrap' not found in backupmeta. "+
-					"you can use \"show config WHERE name='new_collations_enabled_on_first_bootstrap';\" to manually check the config. "+
-					"if you ensure the config 'new_collations_enabled_on_first_bootstrap' in backup cluster is as same as restore cluster, "+
+				"the value 'new_collation_enabled' not found in backupmeta. "+
```
Comment (lance6716) [inline br/pkg/restore/client.go:3692]: can we use the const tidbNewCollationEnabled?

Comment (Defined2014) [reply br/pkg/restore/client.go]: Done @lance6716

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**